sound/soc/sof/topology.c

```diff
@@ -2462,6 +2462,11 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].comp_id = COMP_ID_UNASSIGNED;
 	spcm->stream[SNDRV_PCM_STREAM_CAPTURE].comp_id = COMP_ID_UNASSIGNED;
 
+	INIT_WORK(&spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].period_elapsed_work,
+		  sof_pcm_period_elapsed_work);
+	INIT_WORK(&spcm->stream[SNDRV_PCM_STREAM_CAPTURE].period_elapsed_work,
+		  sof_pcm_period_elapsed_work);
+
```

Comment:
fix commit message

explain why sof_pcm_open() is not called
fix typo "structures"
why should we initialize these structures in dai_load?
what is the impact in non-virtualized cases?